packages/components/src/custom-select-control/index.tsx

```diff
@@ -149,16 +151,14 @@ function CustomSelectControl< T extends CustomSelectOption >(
 			);
 		} );
 
-	const { value: currentValue } = store.getState();
-
 	const renderSelectedValueHint = () => {
 		const selectedOptionHint = options
 			?.map( applyOptionDeprecations )
 			?.find( ( { name } ) => store.getState().value === name )?.hint;
```

Comment:
I think changes on this line from #67733 also need to be reverted.

Reply:
@Mamaduka, in the new PR, these have already been reverted to use currentValue instead of getState().value again. Of course, the way currentValue is determined is slightly different. Does this align with what we were aiming for?

Reply:
Which new PR? Here, I only see the current value reverted on line 161.

Reply:
Got it, I think you're referring to the change that happened on line 157. I'll add a commit to fix it. It would just become 

```js
const renderSelectedValueHint = () => {
		const selectedOptionHint = options
			?.map( applyOptionDeprecations )
			?.find( ( { name } ) => currentValue === name )?.hint;
```

Reply:
Corret! Thanks, @yogeshbhutkar!